llvm/test/Bindings/OCaml/core.ml

```diff
@@ -1115,8 +1126,9 @@ let test_builder () =
   end;
 
   group "metadata"; begin
-    (* CHECK: %metadata = add i32 %P1, %P2, !test !1
-     * !1 is metadata emitted at EOF.
+    (* CHECK: %metadata = add i32 %P1, %P2, !test !{{[0-9]+}}
+     * Number of metadata nodes is not predictable, so we just check for
+     * the presence of metadata here
```

Comment:
I noticed that you made the metadata tests "looser" by checking for a numeric regex instead of the precise metadata ID. As far as I can tell, the tests are still deterministic. Would it be better to continue checking for the exact metadata ID, as the file currently does?

```suggestion
    (* CHECK: %metadata = add i32 %P1, %P2, !test !2
     * Number of metadata nodes is not predictable, so we just check for
     * the presence of metadata here
```